packages/typescript-estree/src/convert.ts

```diff
@@ -3458,4 +3459,20 @@ export class Converter {
 
     throw createError(message, this.ast, start, end);
   }
+  #checkForStatementDeclaration(initializer: ts.ForInitializer): void {
+    if (ts.isVariableDeclarationList(initializer)) {
+      if (
+        !(
+          initializer.flags & ts.NodeFlags.Const ||
+          initializer.flags & ts.NodeFlags.Let ||
+          initializer.flags === ts.NodeFlags.None
+        )
+      ) {
```

Comment:
instead of checking for it "not being one of the valid states", instead can we just check for the one state we want to error on - `(initializer.flags & ts.NodeFlags.Using) !== 0`